src/go/util/httppattern/uri_template_test.go

```diff
@@ -137,10 +134,80 @@ func TestReplaceVariableFieldInUriTemplate(t *testing.T) {
 
 	for _, tc := range testCases {
 		t.Run(tc.desc, func(t *testing.T) {
-			if getUriTemplate, _ := ReplaceVariableFieldInUriTemplate(tc.uriTemplate, tc.varReplace); getUriTemplate != tc.wantUriTemplate {
+			uriTemplate := ParseUriTemplate(tc.uriTemplate)
```

Comment:
There's a lot of tests where ParseUriTemplate is being called and the result being used without checking for `nil`. It works now, but as soon as one test breaks, the tests will panic instead of failing normally.

This is why I would prefer having ParseUriTemplate return an error, as mentioned in a previous PR. Otherwise every caller would have to check for `nil`, very easy to forget. Not going to forget if you have to handle `err` explicitly.

I'm fine with doing this in a follow-up PR. Please double check all production code is checking for nil when using ParseUriTemplate.

Reply:
ACK.